src/traces/sunburst/fx.js

```diff
@@ -217,27 +217,33 @@ module.exports = function attachFxHandlers(sliceTop, entry, gd, cd, opts) {
         var fullLayoutNow = gd._fullLayout;
         var traceNow = gd._fullData[trace.index];
 
-        var clickVal = Events.triggerHandler(gd, 'plotly_' + trace.type + 'click', {
+        var noTransition = isSunburst && (helpers.isHierarchyRoot(pt) || helpers.isLeaf(pt));
+
+        var id = helpers.getPtId(pt);
+        var nextEntry = helpers.isEntry(pt) ?
+            helpers.findEntryWithChild(hierarchy, id) :
+            helpers.findEntryWithLevel(hierarchy, id);
+        var nextLevel = helpers.getPtId(nextEntry);
+
+        var typeClickEvtData = {
             points: [makeEventData(pt, traceNow, opts.eventDataKeys)],
             event: d3.event
-        });
+        };
+        if(!noTransition) typeClickEvtData.nextLevel = nextLevel;
```

Comment:
Wondering if the key could be `nextLevelId` or an object in case we want to provide more info about the node we are transitioning to? E.g.
```js
nextLevel: {
  id: 
  label:
  isEntry:
  isRoot:
}
```

Reply:
Nevermind. It looks we have parts of this info in points.